### PR TITLE
Feat/download

### PR DESCRIPTION
**Pull request by majzabukovnik**

### CODE REVIEW

File: app/Http/Controllers/FileDownloadController.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        if (Storage::exists($filePath)) {
+            $mimeType = Storage::mimeType($filePath);
+
+            // Generate a response to force the file download
+            return response()->stream(function () use ($filePath) {
+                Storage::disk('local')->download($filePath);
+            }, 200, [
+                'Content-Type' => $mimeType,
+                'Content-Disposition' => 'attachment; filename="' . $filename . '"',
```
Comment (majzabukovnik) [inline app/Http/Controllers/FileDownloadController.php:20]: You can change this if you want to have different name of downloaded file form what it was saved on server as

**Comment by majzabukovnik** [comment]
@zanurban Please extensively test download of all kinds of files and from student and teacher accounts. If something doesn't work you can fix it yourself or open a new issue. DO NOT MERGE UNTIL IT WORKS 100%!

**Comment by majzabukovnik** [comment]
Routes for file download are not protected because this somehow breaks functionality.

**Review by zanskorja — CHANGES_REQUESTED** [review]
Donloading of files does not work for me I tried to update composer files
image

**Comment by zanskorja** [comment]
@majzabukovnik Please check if everything works, fix the opened issue and check for register errors (if there are any)

**Review by majzabukovnik — COMMENTED** [review]
I approve, but cannot confirm since I opened this request myself.

**Review by zanskorja — APPROVED** [review]

**Pull request merged**